src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -133,55 +135,80 @@ internal partial class Binder
                         {
                             var typeConstraintSyntax = (TypeConstraintSyntax)syntax;
                             var typeSyntax = typeConstraintSyntax.Type;
-                            if (typeSyntax.Kind() != SyntaxKind.PredefinedType && !SyntaxFacts.IsName(typeSyntax.Kind()))
+                            var typeSyntaxKind = typeSyntax.Kind();
+                            if (typeSyntaxKind != SyntaxKind.PredefinedType && typeSyntaxKind != SyntaxKind.PointerType && !SyntaxFacts.IsName(typeSyntax.Kind()))
```

Comment:
>typeSyntaxKind != SyntaxKind.PointerType [](start = 79, length = 40)

It is not obvious why we are no longer reporting this error for pointers. Could you elaborate?

Reply:
Added a comment. This error is reported for pointers during binding typeSyntax (to prevent cycles). So we don't need a duplicate error here.

---
In reply to: [172416549](https://github.com/dotnet/roslyn/pull/24817#discussion_r172416549) [](ancestors = 172416549)